File: WrecklessEngine/Keyboard.cpp
#include "Keyboard.h"
#include <Windows.h>
#include "HrException.h"
#include "WindowsThrowMacros.h"

static LPDIRECTINPUT8 g_pDirectInput;

namespace Input
{
	bool Keyboard::m_KeyboardState[256] = {};

	bool Keyboard::CapsLock()
	{
		return m_KeyboardState[(int)KeyCode::CapsLock];
	}
	bool Keyboard::NumLock()
	{
		return m_KeyboardState[(int)KeyCode::NumLock];
	}
	bool Keyboard::IsKeyDown(KeyCode key)
	{
		return m_KeyboardState[(int)key];
	}
	bool Keyboard::IsKeyUp(KeyCode key)
	{
		return !IsKeyDown(key);
	}


	void Keyboard::KeyPressed(KeyCode key)
	{
		m_KeyboardState[(int)key] = true;
	}
	void Keyboard::KeyReleased(KeyCode key)
	{
		m_KeyboardState[(int)key] = false;
	}
}

File: WrecklessScript.Core/Input/Keyboard.cs
using System;
using System.Runtime.CompilerServices;

namespace WrecklessScript.Core
{
    namespace Input
    {
        public static class Keyboard
        {
            [MethodImpl(MethodImplOptions.InternalCall)]
            public static extern bool CapsLock();
            [MethodImpl(MethodImplOptions.InternalCall)]
            public static extern bool NumLock();
            [MethodImpl(MethodImplOptions.InternalCall)]
            public static extern bool IsKeyDown(KeyCode key);
            [MethodImpl(MethodImplOptions.InternalCall)]
            public static extern bool IsKeyUp(KeyCode key);
        }
    }
}
